db/context.go

```diff
@@ -64,6 +76,14 @@ func mustGetContextTxn(ctx context.Context) datastore.Txn {
 	return ctx.Value(txnContextKey{}).(datastore.Txn)
 }
 
+// mustGetContextIdentity returns the identity from the context or panics.
+//
+// This should only be called from private functions within the db package
+// where we ensure an identity always exists.
+func mustGetContextIdentity(ctx context.Context) acpIdentity.Identity {
```

Comment:
thought: I don't think we need the same pattern for identity as for transactions. There isn't a case where something is wrong if the identity hasn't been provided. I think we could get away with setting the identity as you do now and getting the identity returning a `NoIdentity` if one isn't present in the context. What do you think?

edit: Come to think of it, we could probably apply the same logic to transactions. We could have just a set and get function here as well with the get returning a new txn if one hasn't been set.

Reply:
I agree, the Identity calls can be simplified. Transactions are a bit more complicated because we need to discard / commit the implicit ones. If we used that pattern for transactions it would be hard to keep track of where to discard / commit.